if you set about:config key 'services.settings.server' to some different url then it will be ignored like:
console.warn: services.settings: Ignoring preference override of remote settings server
console.warn: services.settings: Allow by setting MOZ_REMOTE_SETTINGS_DEVTOOLS=1 in the environment

and it will thus use its embedded one https://firefox.settings.services.mozilla.com/v1 for firefox, and //thunderbird-settings.thunderbird.net/v1 for thunderbird  as defined in firefox-123.0/toolkit/modules/AppConstants.sys.mjs

Index: /var/tmp/portage/www-client/firefox-123.0/work/firefox-123.0/services/settings/Utils.sys.mjs
===================================================================
--- .orig/var/tmp/portage/www-client/firefox-123.0/work/firefox-123.0/services/settings/Utils.sys.mjs
+++ firefox-123.0/services/settings/Utils.sys.mjs
@@ -66,7 +66,7 @@ ChromeUtils.defineLazyGetter(lazy, "allo
   }
 
   if (lazy.gServerURL != AppConstants.REMOTE_SETTINGS_SERVER_URL) {
-    log.warn("Ignoring preference override of remote settings server");
+    log.warn("Ignoring preference override of remote settings server. Will use NOT use the provided aka lazy.gServerURL:'" + lazy.gServerURL + "', which is set in the about:config in 'services.settings.server' but rather the hardcoded aka AppConstants.REMOTE_SETTINGS_SERVER_URL:'" + AppConstants.REMOTE_SETTINGS_SERVER_URL + "'");
     log.warn(
       "Allow by setting MOZ_REMOTE_SETTINGS_DEVTOOLS=1 in the environment"
     );
